src/transformers/modeling_bart.py

```diff
@@ -745,23 +748,23 @@ class LearnedPositionalEmbedding(nn.Embedding):
     position ids are passed to the forward function.
     """
 
-    def __init__(
-        self, num_embeddings: int, embedding_dim: int, padding_idx: int,
-    ):
-        # if padding_idx is specified then offset the embedding ids by
-        # this index and adjust num_embeddings appropriately
+    def __init__(self, num_embeddings: int, embedding_dim: int, padding_idx: int, offset=2):
```

Comment:
This could break backwards compatibiilty for people that used a padding_idx != 1 before.

Also, I think it might be safer and more flexible here to set `offset=None` and add `assert offset is not None` ? What do you think?

Reply:
I'm pretty sure nobody has padding idx =0, because this would have broken for them.

For padding_idx > 1, we can protect by using `offset=padding_idx+1`.

Reply:
Why is offset=None more flexible?

Reply:
In case other models want to use this class, but it's a nit, so not too important to me. Just think it's nicer to "hard-code" values only in the config if possible.

Reply:
I'll make it mandatory